plugins/woocommerce/src/Blocks/OptionsManager.php

```diff
@@ -0,0 +1,37 @@
+<?php
+namespace Automattic\WooCommerce\Blocks;
+
+use Automattic\WooCommerce\Blocks\Options;
+
+/**
+ * OptionsManager class.
+ *
+ * @internal
+ */
+class OptionsManager {
+
+	/**
+	 * Initialization method.
+	 */
+	public function init() {
+		add_action( 'after_switch_theme', array( $this, 'check_should_use_blockified_product_grid_templates' ), 10, 2 );
+	}
+
+	/**
+	 * Checks the old and current themes and determines if the "wc_blocks_use_blockified_product_grid_block_as_template"
+	 * option need to be updated accordingly.
+	 *
+	 * @return void
+	 */
+	public function check_should_use_blockified_product_grid_templates() {
```

Comment:
Do we have any existing tests for this? Seems like something we should write tests for

Reply:
Yup, it's tested as part of the BlockTemplatesUtils here:

https://github.com/woocommerce/woocommerce/blob/38d4a9753627068b8034eb909c9105eee830a4e4/plugins/woocommerce/tests/php/src/Blocks/Utils/BlockTemplateUtilsTest.php#L285-L346